addons/crm_survey/models/survey_user_input.py

```diff
@@ -0,0 +1,171 @@
+                    (==> expected html line: <li>Question — <i>Skipped</i></li>)
+
+            Note: "<br/>&emsp;" is used to get a line break with indentation.
+        """
+        self.ensure_one()
+
+        answers_grouped_by_question = self.user_input_line_ids.grouped('question_id')
+        html_input_lines = []
+        user_nickname, public_user_mail = "", ""
+        for question_id, input_lines in answers_grouped_by_question.items():
+            answer_description, last_row_id = "", ""
+            # Check if the main question is skipped but a comment was submitted
+            if not all(input_line.skipped for input_line in input_lines):
+                for answer_index, answer_id in enumerate(input_lines):
+                    if not answer_id.skipped:
+                        # Check if the question has a nickname recorded or an email answer
+                        if question_id.question_type == "char_box":
+                            if not user_nickname and question_id.save_as_nickname:
+                                user_nickname = answer_id._get_answer_value()
+                            if not public_user_mail and question_id.validation_email:
+                                public_user_mail = answer_id._get_answer_value()
+
+                        # Write description line
+                        if question_id.question_type == "text_box":
+                            answer_description = Markup("<br/>&emsp;%s" % escape(answer_id._get_answer_value()).replace('\n', Markup("<br/>&emsp;")))
+                        elif question_id.question_type == "matrix":
+                            row_id = answer_id.matrix_row_id
+                            if row_id.display_name:
+                                col_value = answer_id.suggested_answer_id.display_name
+                                if last_row_id != row_id:
+                                    last_row_id = row_id
+                                    answer_description += Markup("<br/>&emsp;%s — %s" % (escape(row_id.display_name), escape(col_value)))
+                                else:  # For multiple suggested answers
+                                    answer_description += Markup(", %s" % escape(col_value))
+                            else:  # Comment field
+                                comment_answer = answer_id._get_answer_value()
+                                answer_description += Markup("<br/>&emsp;%s — %s" % (_('Comment'), escape(comment_answer).replace('\n', Markup("<br/>&emsp;"))))
+                        else:
+                            answer = escape(str(answer_id._get_answer_value())).replace('\n', Markup("<br/>&emsp;"))
+                            # For picture answers without label, we use the filename by the way
+                            if answer_id.suggested_answer_id and not answer_id._get_answer_value():
+                                answer = escape(answer_id.suggested_answer_id.value_image_filename)
```

Comment:
Is this the only case we do not have an answer when picking a suggested answer? because it feels strange to read a image filename here. Any other occurence of this logic? :)